cleanup CMakeLists.txt

whatdoineed2do opened the pull request:
Normal users building do not want internal `lib_umskt.so` shared library in a build if they specify `-DBUILD_SHARED_LIBS=0` - this will produce static archives for all internal deps and links a binary that is a normal linux binary (not fully static) and thus can be installed into /usr/local/bin etc
```
mkdir build && cd build
cmake \
    -DCMAKE_BUILD_TYPE=Release \
    -DBUILD_SHARED_LIBS=0 \
    -DUMSKT_USE_SHARED_OPENSSL=1 \
    -DOPENSSL_CRYPTO_LIBRARY=/usr/lib64/libcrypto.so \
    -DOPENSSL_LIBRARY=/usr/lib64/libssl.so \
  ../
make -j
```

This cleans up `CMakeLists.txt` for some redudant params and uses the cmake equivs (no need for static / dynamic library defns etc).

thepwrtank18 commented:
![image](https://github.com/UMSKT/UMSKT/assets/57580668/d1924844-c055-4304-8741-9fafa9acfaa4)
Doesn't show in the PR directly for some reason, but uh...

Neo-Desktop commented:
so the idea here is that the musl static build would be a fully static linux build since a standard glibc build won't ever be fully static

lib_umskt / _umskt.dll is useful for other (GUI) applications which utilize the umskt generation runtime

some of this is still a work in progress, but that's the direction intended

whatdoineed2do commented:
The original intent / generated outputs still exist.

There's some cleanup of various linker items that are not necessary.

But the key item I added (for linux) is to let "normal" users to checkout and build all internal deps as static archives and and image linked against them (and the system glibc) - as a normal user i dont care to have dep on internal `libfmt.so` or `lib_umskt.so`.  This setup is controlled by the `cmake flags` as described above.
```
# on my fedora box
$ ldd umskt 
	linux-vdso.so.1 (0x00007ffd77e7b000)
	libcrypto.so.1.1 => /lib64/libcrypto.so.1.1 (0x00007f511de00000)
	libstdc++.so.6 => /lib64/libstdc++.so.6 (0x00007f511da00000)
	libm.so.6 => /lib64/libm.so.6 (0x00007f511dd24000)
	libgcc_s.so.1 => /lib64/libgcc_s.so.1 (0x00007f511e17d000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f511d600000)
	libz.so.1 => /lib64/libz.so.1 (0x00007f511e163000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f511e1ad000)
```

I've cleaned up the build for:

* windows - openssl installation not available, bumped to 3.1.3 as available
* gjgpp - fixed build (as consequence of above - it was failing on master trying to link the `lib_umskt` as an exe rather than a `.so`)

Neo-Desktop reviewed (approved):
this looks good to me

thepwrtank18 reviewed (approved):
lgtm
